packages/govuk-frontend/package.json

```diff
@@ -25,7 +25,8 @@
   ],
   "sass": "dist/govuk/all.scss",
   "engines": {
-    "node": ">= 4.2.0"
+    "node": "^16.16.0 || ^18.12.0",
+    "npm": "^8.1.0 || ^9.1.0"
   },
```

Comment:
ESLint defaults to Node.js 8 without `engines` and npm only enforces it with [`engine-strict=true`](https://docs.npmjs.com/cli/v9/using-npm/config?v=true#engine-strict)

For context, I only bumped this version because ESLint started shouting 🤦‍♂️ 

Reasons are in the commit, but for us internally (tests, tasks) we have code that failed ESLint on:

* Node.js v14.0.0 for 'fs/promises'
* Node.js v7.6.0 for Async functions

Externally we've shipped [package entry points](https://nodejs.org/docs/latest-v18.x/api/packages.html#package-entry-points) since [GOV.UK Frontend v4.1.0](https://github.com/alphagov/govuk-frontend/releases/tag/v4.1.0) requiring Node.js v12.17.0:

* Node.js v12.16.0 for conditional package exports
* Node.js v12.17.0 for ES modules as `*.mjs`
* Node.js v12.17.0 for subpath package exports
* Node.js v12.20.0 for wildcard package exports

Bundlers go a little bit further back with Webpack 5 required for package exports:

>Webpack 5 requires at least Node.js 10.13.0 (LTS)

They only landed support recently so we could check their Node.js versions?

* https://github.com/webpack/webpack/pull/10953
* https://github.com/rollup/plugins/pull/693
* https://github.com/evanw/esbuild/issues/187

But for initially seemed sensible to pick Node.js v12.20.0+ for pattern/wildcard exports:

```json
"./package.json": "./package.json",
"./*": "./dist/*"
```

## End of life support

What about end of life dates?

Node.js v12 LTS, 30 April 2022
Node.js v14 LTS, 30 April 2023
Node.js v16 LTS, 11 Sep 2023
Node.js v18 LTS, 30 Apr 2025

Reply:
To clarify, as well, that's the shipping of that updated field that worries me, not having it in the repo (as it's neat to  clarify which version of Node our tools use). Wondering if there'd be a way to remove it before shipping, for example with a combination of:

- `npm pkg delete engines --workspace govuk-frontend` before the package is built (prepack)
- `git checkout packages/govuk-frontend/package.json` to restore it once the package is built (postpack or is it too early?)

Reply:
Just pushed up https://github.com/alphagov/govuk-frontend/pull/3498/commits/4c2d383d526781e3fde3b8c10ac3cbaea9c290d5

I've configured ESLint with Node.js `^18.12.0` but restored `>= 4.2.0` to the package

Reply:
Ace, that makes me more confident we're not breaking anything for our users there. Thanks! ⭐